### ISSUE TITLE
Topics in index should link to most recently read page, rather than first page

### ISSUE THREAD

**Issue by timdiggins** [thredded/thredded#362]
Given that we store the page of the most recently read page, shouldn't we be linking to this page in the topics#index ?

**Comment by glebm** [comment]
I think we are. It also works for me on https://thredded.org/thredded/main-board.

**Comment by timdiggins** [comment]
@glebm agreed, have understood where it happens in the view model now. 

I was thinking (in addition to linking to the page as we are doing) we should link to the anchor of the last read post if possible. This anchor should be calculated stored in UserTopicReadState.

**Comment by timdiggins** [comment]
from #533: 

> And I think it would be good to highlight the unread posts when you first view them (when they are transitioning from read-> unread as it were), like we highlight unread topics

@jayroh I know you've had some thoughts about this -- maybe about doing it in Javascript w/scrollspy for example ( I'm less keen on this, even though it is less accurate, seems like it could be nightmarish to support)

**Comment by jayroh** [comment]
@timdiggins Yeah your estimation is right on. This is the "best" way to track all of this but it's no easy task as there's a lot to juggle. On topics#show - the topic, page, and post

Then of course there's topics#index - marking read or un-read on all of those.

JS is the best way to do this and in terms of support, as long as it's well tested it should be relatively manageable.

**Comment by timdiggins** [comment]
I'm in favour of a simple approach as the next step:

* when viewer views (requests the html of) the last page of posts , then mark all these as read  (via controller) the last page of posts (as roughly as present -- though we ignore what page is being read)

* add a marker saying "unread posts below here" in the gap between read and unread posts

* when linking to a topic page, link to: the last post (specific `#post-2312` anchor) if all read, or to the top of the page (no anchor) if none read, or to the "unread posts marker" (`#unread-posts` anchor) if some but not all are read.

**Comment by timdiggins** [comment]
decision on https://github.com/thredded/thredded/issues/535 is relevant to putting a "unread posts below here" in the UI.

**Comment by glebm** [comment]
> I'm in favour of a simple approach as the next step:

+1

> add a marker saying "unread posts below here" in the gap between read and unread posts

#536 is a pre-requisite for adding the marker, so I'll merge it tomorrow if there are no objections.

> when linking to a topic page, link to: the last post (specific #post-2312 anchor) if all read, or to the top of the page (no anchor) if none read, or to the "unread posts marker" (#unread-posts anchor) if some but not all are read.

+1
